### DISCUSSION TITLE
Fortran - CGNS Parallel Output in Polyhedral Mesh

### DISCUSSION THREAD

**Discussion by SpirosZafeiris** [CGNS/CGNS#366] [CGNStalk]
Dear All,

I am trying to write a Parallel CGNS output routine for an in-house Fortran90 - MPI solver. I have already implemented a routine for fixed size elements which works.

Now, I would like to extend to polyhedral meshes and it seems that CGNS supports only fixed-size elements (again only in parallel).

Is there a way to circumvent this problem or will there be any CGNS poly-element support in the future?

Thanks.

**Comment by mennodeij** [comment]
Since release 4.3.0 there are the functions
```
cgp_poly_section_write
cgp_poly_elements_write_data
```
that allow writing of polygonal and polyhedral sections in parallel. Reading is, I think, not possible in parallel though.

**Reply by SpirosZafeiris** [reply]
Thank you for your quick response. 

I run the latest version of the library which was git-cloned and compiled. However, I get an undefined reference to `cgp_poly_section_write_f` function call (the `cgp_poly_elements_write_data_f` call is recognized by the compiler).

Additionally, in the function call list of the following documentation site [https://cgns.github.io/cgns-modern.github.io/standard/MLL/CGNS_PMLL_library.html#element](url) (element connectivity section), this particular routine is only mentioned for C (and not for Fortran).

Thank you again.

**Comment by mennodeij** [comment]
Ok, looks like that function is missing its Fortran interface, see https://cgnsorg.atlassian.net/browse/CGNS-276 
You could add the required entry/entries to `cg_ftoc.c` and `cgns_f.F90`

**Comment by brtnfld** [comment]
I have them implemented in my fork: https://github.com/brtnfld/CGNS/tree/CGNS-276. I can push the branch to the main repo.
I need to write tests from them before they are pushed to develop, so they are untested.

**Comment by PhDFlo** [comment]
I think this is related to this issue: https://cgnsorg.atlassian.net/browse/CGNS-276
